docs/react-v9/contributing/dev-env.md

```diff
@@ -1,4 +1,100 @@
-// TODO #30642 - A doc explaining how to set up the dev environment.
+This document describes how to set up your development environment and contribute changes to the Fluent UI project. It assumes basic working knowledge with Git and related tools. A typical developer should budget 2 hours from fresh install to their first successful build completion.
 
-Assignee - Mason
-Reviewer - Mak
+- [Basic setup](#basic-setup)
+  - [Install dev tools](#install-dev-tools)
+  - [Verify your environment](#verify-your-environment)
+  - [Building (without contributing)](#building-without-contributing)
+- [Configuration for contribution](#Configuration-for-contribution)
+  - [Creating your own fork](#creating-your-own-fork)
+  - [Cloning your fork and setting the upstream remote](#cloning-your-fork-and-setting-the-upstream-remote)
+  - [Git configuration](#git-configuration)
+
+# Basic setup
+
+### Install dev tools
+
+- If you don't have a **GitHub account**, [create one](https://github.com/join)
+  - Microsoft employees: please [link your GitHub account](https://repos.opensource.microsoft.com) (new or existing) to your MS account
+- Install **[Node.js LTS](https://nodejs.org/en/)** (18 as of writing)
+  - Optional: If you're developing across multiple repos with varying Node version requirements, you may want to use `nvm` to install and manage Node versions.
+- Install **[Yarn 1](https://classic.yarnpkg.com/) (we do not support Yarn 2)**
+  - easiest way: run `npm install -g yarn@1`
+- Install **[Git](https://git-scm.com/)**
+- Install **[Visual Studio Code](https://code.visualstudio.com/)** or any other editor of your preference
+- Optional: If you'd like a GUI for Git, some team members use **[GitHub app](https://desktop.github.com/)** or **[SourceTree](https://www.atlassian.com/software/sourcetree)**
+- Optional: **[Snipping tool](https://apps.microsoft.com/detail/9mz95kl8mr0l?launch=true&mode=full&hl=en-us&gl=us&ocid=bingwebsearch)** for screen grabs and recordings
+- Optional: **[Accessibility insight for web](https://accessibilityinsights.io/)**
+- Optional: **[Node Version Manager](https://github.com/nvm-sh/nvm)**
+- Optional: **[CSpell VS Code Extension](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker)**
```

Comment:
I'll add this to our vscode recommended extension setup

Reply:
added